### PR TITLE
refactor: scaling the signals using minmax logic

### PR DESCRIPTION
**Pull request by gcroci2**

### CODE REVIEW

File: annubes/task.py [hunk 1]
```diff
@@ -165,42 +162,32 @@ def generate_trials(
...
-        """Rescale `input_` value along (0,`coeff`) if `coeff` is non-zero.
+        """Rescale `input_` array to a given range.
 
         Rescaling happens as follows:
-            coeff * (input_ - min_intensity) / (max_intensity - min_intensity)
+
+            `X_std = (input_ - input_.min()) / (input_.max() - input_.min())`
+            `X_scaled = X_std * (max - min) + min`
+            where min, max = range.
+        The logic is the same as that of `sklearn.preprocessing.MinMaxScaler` estimator. Each array is rescaled to the
```
Comment (DaniBodor) [inline annubes/task.py:177]: Isn't this function the same as sklearn.preprocessing.minmax_scale()`? Why not just import that [function](https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.minmax_scale.html#sklearn.preprocessing.minmax_scale) rather than defining our own version of it?

File: annubes/task.py [hunk 2]
```diff
@@ -165,42 +162,32 @@ def generate_trials(
...
 
-    def _rescale(
+    def _minmaxscaler(
         self,
-        input_: float,
-        coeff: float,
-        min_intensity: float | None = None,
-        max_intensity: float | None = None,
+        input_: NDArray[np.float32],
+        input_range: tuple[float, float] = (0, 1),
```
Comment (DaniBodor) [inline annubes/task.py:168]: I think `input_range` is a bit confusing nomenclature. I believe you mean it's the range of the input data of `Task` after rescaling. But it sounds like it's the range of the input to this method.
Perhaps we can call it `rescale_range` instead?
```suggestion
        rescale_range: tuple[float, float] = (0, 1),
```

File: annubes/task.py [hunk 3]
```diff
@@ -165,42 +162,32 @@ def generate_trials(
...
+        given range, for each trial contained in `input_`.
+
 
         Args:
-            input_: Value that will be rescaled.
-            coeff: Maximum value after rescaling. If set to 0, `input_` is returned unmodified.
-            min_intensity: Minimum value of the input intensities. Defaults to `min(self.stim_intensities)`
-            max_intensity: Maximum value of the input intensities. Defaults to `max(self.stim_intensities)`
+            input_ (NDArray[np.float32]): Input array of shape (self._ntrials, len(self.time), self.n_inputs).
+            input_range (tuple[float, float]): Desired range of transformed data. Defaults to (0, 1).
```
Comment (DaniBodor) [inline annubes/task.py:183]: I believe the types don't need to be mentioned in the docstring as long as they are type hinted (see the `def function_with_pep484_type_annotations` example [here](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html))
```suggestion
            input_: Input array of shape (self._ntrials, len(self.time), self.n_inputs).
            input_range: Desired range of transformed data. Defaults to (0, 1).
```

File: annubes/task.py [hunk 4]
```diff
@@ -243,30 +230,26 @@ def _build_trials_inputs(self) -> NDArray[np.float32]:
...
             for idx, m in enumerate(self.modalities):
                 if (self._modality_seq[n] != "catch") and (m in self._modality_seq[n]):
-                    sel_value_in[n, idx] = self._rng.choice(self.stim_intensities[1:], 1)
-                sel_value_in[n, idx] = self._rescale(sel_value_in[n, idx], self.rescaling_coeff)
-                x[n, self._phases["input"], idx] = sel_value_in[n, idx]
-                x[n, self._phases["fix_time"], idx] = self._rescale(
-                    self.fix_intensity,
-                    self.rescaling_coeff,
-                )
+                    value = self._rng.choice(self.stim_intensities, 1)
```
Comment (DaniBodor) [inline annubes/task.py:237]: when would the second part of that if statement not be true?

File: annubes/task.py [hunk 5]
```diff
@@ -243,30 +230,26 @@ def _build_trials_inputs(self) -> NDArray[np.float32]:
...
+                    value = 0
+                x[n, self._phases["input"], idx] = value
+                x[n, self._phases["fix_time"], idx] = self.fix_intensity
             x[n, self._phases["input"], self.n_inputs - 1] = 1  # start cue
 
         # generate noise
         alpha = self.dt / self.tau
         noise_factor = self.noise_std * np.sqrt(2 * alpha) / alpha
-        noise = noise_factor * self._rng.normal(loc=0, scale=1, size=x.shape)
+        x += noise_factor * self._rng.normal(loc=0, scale=1, size=x.shape)
```
Comment (DaniBodor) [inline annubes/task.py:247]: Should we add a check somewhere that neither `dt` nor `tau` are 0 and return a better warning than the one that would be returned here if that would happen?

Also, maybe we can change the comment on line 244 to `add noise`.

**Review by DaniBodor — CHANGES_REQUESTED** [review]
Much better to rescale after setting x.
Shall we use sklearn for rescaling rather than defining our own method for it (see my comment below)?

File: annubes/task.py (hunk 1 above)
Comment (gcroci2) [reply annubes/task.py:177]: Because the axis used by sklearn are different (axis=0); here we have an array with shape (number of trials, number of data points in one time series, number of time series). This is not the shape that MinMaxScaler expects, which has been developed for standard tabular data (e.g., shapes like n_samples, n_features). In the beginning, this was my idea as well (i.e., import directly scikit-learn functions), but I couldn't find a straightforward way to actually do it. Feel free to try yourself of course!! It would be definitely better to just reuse an already-existing function

Comment (DaniBodor) [reply annubes/task.py:177]: I understand.

File: annubes/task.py (hunk 3 above)
Comment (gcroci2) [reply annubes/task.py]: Makes total sense!

File: annubes/task.py (hunk 4 above)
Comment (gcroci2) [reply annubes/task.py]: Never. I removed it :)
